packages/cli/test/integration/repositories/execution.repository.test.ts

```diff
@@ -0,0 +1,209 @@
+import config from '@/config';
+import * as Db from '@/Db';
+
+import * as testDb from '../shared/testDb';
+import type { ExecutionStatus } from 'n8n-workflow';
+import { LoggerProxy } from 'n8n-workflow';
+import { getLogger } from '@/Logger';
+import type { ExecutionRepository } from '../../../src/databases/repositories';
+import type { ExecutionEntity } from '../../../src/databases/entities/ExecutionEntity';
+
+describe('ExecutionRepository.prune()', () => {
+	const now = new Date();
+	const yesterday = new Date(Date.now() - 24 * 60 * 60 * 1000);
+	let executionRepository: ExecutionRepository;
+	let workflow: Awaited<ReturnType<typeof testDb.createWorkflow>>;
+
+	beforeAll(async () => {
+		LoggerProxy.init(getLogger());
+		await testDb.init();
+
+		const { Execution } = Db.collections;
+
+		executionRepository = Execution;
+		workflow = await testDb.createWorkflow();
+	});
+
+	beforeEach(async () => {
+		await testDb.truncate(['Execution']);
+	});
+
+	afterAll(async () => {
+		await testDb.terminate();
+	});
+
+	async function findAllExecutions() {
+		return Db.collections.Execution.find({
+			order: { id: 'asc' },
+			withDeleted: true,
+		});
+	}
+
+	describe('when EXECUTIONS_DATA_PRUNE_MAX_COUNT is set', () => {
+		beforeAll(() => {
+			config.set('executions.pruneDataMaxCount', 1);
+			config.set('executions.pruneDataMaxAge', 336);
+		});
```

Comment:
We could reset oclif state after each test at the top level: `config.load(config.default);` As we might later want to add some tests here that are unrelated to pruning.